Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -195,66 +195,50 @@
-%   E.g., for a 8 echoes Hadamard acquisition, a single control or PLD-encoded acquisition is repeated 8 times, so:
+% INNER REPETITIONS
+% Normally, for a 4-block Hadamard, we acquire 4 volumes. But there may be readout-related reasons for having more volumes than that.
+% E.g., with a multi-TE readout, each Hadamard-encoded volume is acquired multiple times equal to the number of echoes. Because this is repeated 
+% at the readout level, we define these here as "inner repetitions". E.g., for a 8-echoes Hadamard acquisition, a single control or PLD-encoded acquisition is repeated 8 times.
 %
 % innerRepetition: single-volume repetitions within a Hadamard block
 
 [~, ~, indexUniquePLD] = unique(xQ.Initial_PLD); % Find volume indices of unique PLDs
-indexFirstPLD = find(indexUniquePLD==1); % Find the volumes of the first PLD
-indexSecondPLD = find(indexUniquePLD==2); % Find the volumes of the second PLD
+indexSecondPLD = find(indexUniquePLD==2); % Find the volumes of the second PLD - all PLDs before the second PLD are equivalent
 if isempty(indexSecondPLD)
 	nInnerRepetitions = length(xQ.Initial_PLD); % There's no second PLD and nInnerRepetitions is the length of the PLD vector
 else
-	nInnerRepetitions = indexSecondPLD - 1; % The number of first-PLDs, i.e. the count of all PLDs before the second one, is the number of inner repetitions
+	nInnerRepetitions = indexSecondPLD(1) - 1; % The number of first-PLDs, i.e. the count of all PLDs before the second PLD is the number of inner repetitions
 end
 
 if nInnerRepetitions ~= xQ.NumberEchoTimes % Check for TE number differ from the number of inner repeats as this is a special option needing a special reordering
-	warning('Number of inner repetitions and echo-times does not match, such a sequence is not yet implemented');
+	warning('Number of inner repetitions and echo-times does not match for a Hadamard sequence. Quantification of such a sequence is not yet implemented');
 end
 
-nVolumesPerBlock = nEncodedVolumes / nBlocks;
-
 % We define how many volumes are within an inner repetition accounting for the number of blocks within a Hadamard volume
 nVolumesPerInnerRepetition = nBlocks * nInnerRepetitions;
 
-% We calculate the number of full repetitions - that is repeating the entire acquisition scheme
+% OUTER REPETITIONS
+% The full Hadamard-encoded combination of ASL measurements can be repeated for averaging to increase SNR. 
+% Though note that some readouts like 3D GRASE are typically segmented to reduce the acquisition PSF in the Z-direction, 
+% increasing the acquisition of a single control or label volume with a factor of 2, 4, 8 (depending on the number of segments).
+% Hence, for 3D Grase Hadamard, there's typically only clinically scanning time to scan 1-3 repetitions, which we here define as "outer repetitions",
+% 
+% We calculate the number of full repetitions - that is repeating the entire acquisition scheme (also can be called outer repetitions
```

Comment:
Can you give in the comment of section 4 , the example for 8 echoes and 4 PLDs?
Now it has the example of 8 PLDs and 4 echoes. Same throughout.